### PR TITLE
Use prettier arrow in stack traces (to highlight the line with the error)

### PR DESCRIPTION
**Pull request by holly-cummins**
## TLDR: the 'after'

*EDIT:* After testing on Windows, I've changed the arrow. 
 
image

## What problem were we solving?

In @phillip-kruger's recent demo of the advanced exception context, I noticed that we use an ascii-art arrow. 
image

"Hmm," I thought. "With so many nice arrows in the unicode spec these days, why don't we use one of them instead?" With hindsight, I should have asked @phillip-kruger if he’d tried this already, but I thought “I won’t bother him with a conversation for a one-character change, I’ll just do the PR and we can discuss it there.” Then I started testing, and realised why Philip maybe opted for the ascii art. But by then it was too late, and I was invested.

image

There are two risks with this change: 
- Will the character be available in the terminal and browser font on the developer's system?
- Can we make the code in the line with the arrow align properly with the rest?

It turns out both are a problem, but I think I got there in the end. 

### MacOS

image

image

### Windows 

image

image

### Linux 

image

image

### Character width issues 

Even on a single machine, I got character-alignment problems. It turns out that fixed-width fonts do not have a consistent width for the arrows. It will presumably vary by OS, and it also varies between browser and terminal. I liked the heavy arrow ⮕, but that was subtly misaligned in the browser:

image

... and hopelessly misaligned in the terminal:

image

⟶ is nice and prominent, perfect in the console, 

image

But terrible in the browser

image

In my experiments, ⇨ works in the browser:

image

and also the terminal 

image

It's a bit more visible than ⇾, which also worked for me in terms of width:

image

image

### Character set issues

Once I found an arrow that rendered well in both terminal and browser on Mac, we tried it on Windows, and ... no good.

image

So I went back to the drawing board. 

## How to test 

Here's some text which can be pasted into a terminal to test on other OSes:

```
Exception in GreetingResource.java:14
          12      @Produces(MediaType.TEXT_PLAIN)
          13      public String hello() {
        ⇨ 14          if (Math.random( )> 0.01) { throw new RuntimeException("test");}
          15          return "Hello from Quarkus RESTyes";
          16      }
```

(Testing in the browser would need a build of the patch.)

## Other options 
For reference, here are all the arrows I considered, along with a `.` to show alignment. Only the fourth one works on Windows:

⟶ .
⭢ .
⮕ .
→ .
⇨ .
⇢ .
⇾ .

### CODE REVIEW

**Review by phillip-kruger — APPROVED** [review]

**Comment by phillip-kruger** [comment]
Very cool !! Thanks. I did not think of that at all :)

**Comment by holly-cummins** [comment]
@gastaldi has kindly tested on Windows, and concerns were justified:
image

**Review by gastaldi — APPROVED** [review]

**Comment by gastaldi** [comment]
Here is a screenshot in Edge (Windows 11): 

image
